store/redis.go

```diff
@@ -0,0 +1,96 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// RedisConfig ... user configurable
+type RedisConfig struct {
+	Endpoint string
+	Password string
+	DB       int
+	Eviction time.Duration
+	Profile  bool
+}
+
+// RedStore ... Redis storage backend implementation
+type RedStore struct {
+	eviction time.Duration
+
+	client *redis.Client
+
+	profile bool
+	reads   int
+	entries int
+}
+
+var _ PrecomputedKeyStore = (*RedStore)(nil)
+
+// NewRedisStore ... constructor
+func NewRedisStore(cfg *RedisConfig) (*RedStore, error) {
+	client := redis.NewClient(&redis.Options{
+		Addr:     cfg.Endpoint,
+		Password: cfg.Password,
+		DB:       cfg.DB,
+	})
+
+	// ensure server can be pinged using potential client connection
+	cmd := client.Ping(context.Background())
+	if cmd.Err() != nil {
+		return nil, cmd.Err()
+	}
+
+	return &RedStore{
+		eviction: cfg.Eviction,
+		client:   client,
+		profile:  cfg.Profile,
+		reads:    0,
+	}, nil
+}
+
+// Get ... retrieves a value from the Redis store. Returns nil if the key is not found vs. an error
+// if the key is found but the value is not retrievable.
+func (r *RedStore) Get(ctx context.Context, key []byte) ([]byte, error) {
+	value, err := r.client.Get(ctx, string(key)).Result()
+	if errors.Is(err, redis.Nil) { // key DNE
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	if r.profile {
+		r.reads++
```

Comment:
this will break if Get is called concurrently. Why are we not using prometheus metrics instead of these custom stats? prom counters/gauges use atomic ints precisely because of this.

Reply:
it won't get called concurrently though and if it ever does we can introduce atomic reference types for both the client and these stats fields. this is done purely for integration testing.

Reply:
I personally don't like to have any code in main code that is for testing purposes only. Feel like we should use prometheus metrics for this, and this would give us the added benefit that we could also export them?
Although its fine to not do this now, feel free to resolve.

Reply:
agree your proposed approach is a better solution long-term - can file an issue and close out

Reply:
https://github.com/Layr-Labs/eigenda-proxy/issues/113